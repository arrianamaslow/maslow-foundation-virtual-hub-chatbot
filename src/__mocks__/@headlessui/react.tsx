/**
 * @jest-environment jsdom
 */

import { DialogProps } from '@headlessui/react'
import React, { ReactNode } from 'react'

type DialogPanelTestProps = {
  children?: ReactNode
}

interface DialogTestProps extends DialogProps {
  children?: ReactNode
}

const DialogPanel: React.FC<DialogPanelTestProps> = ({ children }) => {
  return <div data-testid='mock-dialog-panel-component'>{children}</div>
}

const Dialog: React.FC<DialogTestProps> = ({ open, children }) => {
  return (
    <div data-testid='mock-dialog-component'>
      {open ? 'DIALOGUEOPENED' : 'DIALOGUECLOSED'}
      {children}
    </div>
  )
}

export { Dialog, DialogPanel }
