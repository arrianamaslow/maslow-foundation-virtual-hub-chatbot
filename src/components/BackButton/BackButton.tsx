import { useRouter } from 'next/navigation'
import React from 'react'

export type BackButtonProps = {
  href: string
}

const BackButton: React.FC<BackButtonProps> = ({ href }) => {
  const router = useRouter()
  return (
    <button
      data-testid='back-button'
      onClick={() => {
        router.push(href)
      }}
      className='h-6 w-16'
    >
      <div className='flex flex-row'>
        <img
          className='border-[1.5px] border-black rounded-lg py-[6px] w-6 h-6'
          src='/img/back-arrow.svg'
        />
        <span className='flex justify-center text-base pl-[7px] font-baskervald'>Back</span>
      </div>
    </button>
  )
}
export default BackButton
