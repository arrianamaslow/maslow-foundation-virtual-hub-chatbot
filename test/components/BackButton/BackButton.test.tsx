/**
 * @jest-environment jsdom
 */

import { fireEvent, render, screen } from '@testing-library/react'
import BackButton from '@/components/BackButton/BackButton'
import '@testing-library/jest-dom'

const pushMock = jest.fn()
jest.mock('next/navigation', () => ({
  useRouter: jest.fn().mockImplementation(() => {
    return { push: pushMock }
  })
}))

describe('<BackButton />', () => {
  it('renders correctly', () => {
    const tree = render(<BackButton href='/chat' />)
    expect(tree).toMatchSnapshot()
  })

  beforeEach(() => {
    render(<BackButton href='/chat' />)
  })

  it('properly passes the href to its link sub component', async () => {
    const backButton = screen.getByTestId('back-button')
    fireEvent.click(backButton)
    expect(pushMock).toHaveBeenCalledWith('/chat')
  })

  it('renders the link sub component', () => {
    const backButton = screen.getByTestId('back-button')
    expect(backButton).toBeInTheDocument()
    expect(backButton).toBeEnabled()
  })
})
