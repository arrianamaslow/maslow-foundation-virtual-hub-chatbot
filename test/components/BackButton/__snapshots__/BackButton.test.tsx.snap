// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackButton /> renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="h-6 w-16"
        data-testid="back-button"
      >
        <div
          class="flex flex-row"
        >
          <img
            class="border-[1.5px] border-black rounded-lg py-[6px] w-6 h-6"
            src="/img/back-arrow.svg"
          />
          <span
            class="flex justify-center text-base pl-[7px] font-baskervald"
          >
            Back
          </span>
        </div>
      </button>
    </div>
    <div>
      <button
        class="h-6 w-16"
        data-testid="back-button"
      >
        <div
          class="flex flex-row"
        >
          <img
            class="border-[1.5px] border-black rounded-lg py-[6px] w-6 h-6"
            src="/img/back-arrow.svg"
          />
          <span
            class="flex justify-center text-base pl-[7px] font-baskervald"
          >
            Back
          </span>
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="h-6 w-16"
      data-testid="back-button"
    >
      <div
        class="flex flex-row"
      >
        <img
          class="border-[1.5px] border-black rounded-lg py-[6px] w-6 h-6"
          src="/img/back-arrow.svg"
        />
        <span
          class="flex justify-center text-base pl-[7px] font-baskervald"
        >
          Back
        </span>
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
